types/binary_literal_test.go

```diff
@@ -15,271 +15,280 @@
 
-type testBinaryLiteralSuite struct {
-}
+	t.Run("TestParseBitStr", func(t *testing.T) {
+		t.Parallel()
+		tbl := []struct {
+			Input    string
+			Expected []byte
+			IsError  bool
+		}{
+			{"b''", []byte{}, false},
+			{"B''", []byte{}, false},
+			{"0b''", nil, true},
+			{"0b0", []byte{0x0}, false},
+			{"b'0'", []byte{0x0}, false},
+			{"B'0'", []byte{0x0}, false},
+			{"0B0", nil, true},
+			{"0b123", nil, true},
+			{"b'123'", nil, true},
+			{"0b'1010'", nil, true},
+			{"0b0000000", []byte{0x0}, false},
+			{"b'0000000'", []byte{0x0}, false},
+			{"B'0000000'", []byte{0x0}, false},
+			{"0b00000000", []byte{0x0}, false},
+			{"b'00000000'", []byte{0x0}, false},
+			{"B'00000000'", []byte{0x0}, false},
+			{"0b000000000", []byte{0x0, 0x0}, false},
+			{"b'000000000'", []byte{0x0, 0x0}, false},
+			{"B'000000000'", []byte{0x0, 0x0}, false},
+			{"0b1", []byte{0x1}, false},
+			{"b'1'", []byte{0x1}, false},
+			{"B'1'", []byte{0x1}, false},
+			{"0b00000001", []byte{0x1}, false},
+			{"b'00000001'", []byte{0x1}, false},
+			{"B'00000001'", []byte{0x1}, false},
+			{"0b000000010", []byte{0x0, 0x2}, false},
+			{"b'000000010'", []byte{0x0, 0x2}, false},
+			{"B'000000010'", []byte{0x0, 0x2}, false},
+			{"0b000000001", []byte{0x0, 0x1}, false},
+			{"b'000000001'", []byte{0x0, 0x1}, false},
+			{"B'000000001'", []byte{0x0, 0x1}, false},
+			{"0b11111111", []byte{0xFF}, false},
+			{"b'11111111'", []byte{0xFF}, false},
+			{"B'11111111'", []byte{0xFF}, false},
+			{"0b111111111", []byte{0x1, 0xFF}, false},
+			{"b'111111111'", []byte{0x1, 0xFF}, false},
+			{"B'111111111'", []byte{0x1, 0xFF}, false},
+			{"0b1101000011001010110110001101100011011110010000001110111011011110111001001101100011001000010000001100110011011110110111100100000011000100110000101110010", []byte("hello world foo bar"), false},
+			{"b'1101000011001010110110001101100011011110010000001110111011011110111001001101100011001000010000001100110011011110110111100100000011000100110000101110010'", []byte("hello world foo bar"), false},
+			{"B'1101000011001010110110001101100011011110010000001110111011011110111001001101100011001000010000001100110011011110110111100100000011000100110000101110010'", []byte("hello world foo bar"), false},
+			{"0b01101000011001010110110001101100011011110010000001110111011011110111001001101100011001000010000001100110011011110110111100100000011000100110000101110010", []byte("hello world foo bar"), false},
+			{"b'01101000011001010110110001101100011011110010000001110111011011110111001001101100011001000010000001100110011011110110111100100000011000100110000101110010'", []byte("hello world foo bar"), false},
+			{"B'01101000011001010110110001101100011011110010000001110111011011110111001001101100011001000010000001100110011011110110111100100000011000100110000101110010'", []byte("hello world foo bar"), false},
+		}
+		for _, item := range tbl {
+			b, err := ParseBitStr(item.Input)
+			if item.IsError {
+				require.NotNil(t, err, fmt.Sprintf("%#v", item))
```

Comment:
```suggestion
				require.Error(t, err)
```

ditto others. testify already provides us rendered error message.

Reply:
Done